### ISSUE TITLE
2017.7.6 - Exiting command with CTRL-C gives Error/Traceback

### ISSUE THREAD

**Issue by frogunder** [saltstack/salt#47901]
### Description of Issue/Question
On 2017.7.6 when using CTRL-C to exit running command it gives this traceback:
```
[root@frode1 ~]# salt '*' test.sleep 30
^C[ERROR   ] An un-handled exception was caught by salt's global exception handler:
Exception: DummyFuture does not support blocking for results
Traceback (most recent call last):
  File "/usr/bin/salt", line 10, in <module>
    salt_main()
  File "/usr/lib/python2.7/site-packages/salt/scripts.py", line 481, in salt_main
    client.run()
  File "/usr/lib/python2.7/site-packages/salt/cli/salt.py", line 173, in run
    for full_ret in cmd_func(**kwargs):
  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 831, in cmd_cli
    self.event.close_pub()
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 391, in close_pub
    self.subscriber.close()
  File "/usr/lib/python2.7/site-packages/salt/transport/ipc.py", line 756, in close
    self._read_sync_future.exception()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 230, in exception
    self._check_done()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 293, in _check_done
    raise Exception("DummyFuture does not support blocking for results")
Exception: DummyFuture does not support blocking for results
Traceback (most recent call last):
  File "/usr/bin/salt", line 10, in <module>
    salt_main()
  File "/usr/lib/python2.7/site-packages/salt/scripts.py", line 481, in salt_main
    client.run()
  File "/usr/lib/python2.7/site-packages/salt/cli/salt.py", line 173, in run
    for full_ret in cmd_func(**kwargs):
  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 831, in cmd_cli
    self.event.close_pub()
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 391, in close_pub
    self.subscriber.close()
  File "/usr/lib/python2.7/site-packages/salt/transport/ipc.py", line 756, in close
    self._read_sync_future.exception()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 230, in exception
    self._check_done()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 293, in _check_done
    raise Exception("DummyFuture does not support blocking for results")
Exception: DummyFuture does not support blocking for results
```

It should return this info ( as it does on 2017.7.5):
```
[root@frode1 ~]# salt '*' test.sleep 30
^C
Exiting gracefully on Ctrl-c
This job's jid is: 20180530154948307971
The minions may not have all finished running and any remaining minions will return upon completion. To look up the return data for this job later, run the following command:

salt-run jobs.lookup_jid 20180530154948307971
```

### Setup
Install 2017.7.6

### Steps to Reproduce Issue
run a command and CTRL-C before it finishes.

### Versions Report
```
Salt Version:
           Salt: 2017.7.6
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.28.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.1
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

**Comment by rallytime** [comment]
@DmitryKuzmenko I think this was caused by some of the changes needed for the tornado changes. This doesn't need to block the upcoming 2017.7.6 release, but it would be good to get fixed for the next release in 2017.7.7.

**Comment by frogunder** [comment]
I also see the same thing on 2018.3.1

```
[root@frode1 ~]# salt '*' test.sleep 30
^C[ERROR   ] An un-handled exception was caught by salt's global exception handler:
Exception: DummyFuture does not support blocking for results
Traceback (most recent call last):
  File "/usr/bin/salt", line 10, in <module>
    salt_main()
  File "/usr/lib/python2.7/site-packages/salt/scripts.py", line 485, in salt_main
    client.run()
  File "/usr/lib/python2.7/site-packages/salt/cli/salt.py", line 193, in run
    for full_ret in cmd_func(**kwargs):
  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 842, in cmd_cli
    self.event.close_pub()
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 398, in close_pub
    self.subscriber.close()
  File "/usr/lib/python2.7/site-packages/salt/transport/ipc.py", line 755, in close
    self._read_sync_future.exception()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 230, in exception
    self._check_done()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 293, in _check_done
    raise Exception("DummyFuture does not support blocking for results")
Exception: DummyFuture does not support blocking for results
Traceback (most recent call last):
  File "/usr/bin/salt", line 10, in <module>
    salt_main()
  File "/usr/lib/python2.7/site-packages/salt/scripts.py", line 485, in salt_main
    client.run()
  File "/usr/lib/python2.7/site-packages/salt/cli/salt.py", line 193, in run
    for full_ret in cmd_func(**kwargs):
  File "/usr/lib/python2.7/site-packages/salt/client/__init__.py", line 842, in cmd_cli
    self.event.close_pub()
  File "/usr/lib/python2.7/site-packages/salt/utils/event.py", line 398, in close_pub
    self.subscriber.close()
  File "/usr/lib/python2.7/site-packages/salt/transport/ipc.py", line 755, in close
    self._read_sync_future.exception()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 230, in exception
    self._check_done()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 293, in _check_done
    raise Exception("DummyFuture does not support blocking for results")
Exception: DummyFuture does not support blocking for results
[root@frode1 ~]# salt --version
salt 2018.3.1 (Oxygen)
```

**Comment by rallytime** [comment]
@frogunder This should be fixed with #48185. Can you give that a try when you have a moment?

**Comment by frogunder** [comment]
It is working correctly for me with that fix.
